.github/workflows/scorecard.yml

```diff
@@ -0,0 +1,58 @@
+name: Scorecard supply-chain security
+on:
+  # For Branch-Protection check. Only the default branch is supported. See
+  # https://github.com/ossf/scorecard/blob/main/docs/checks.md#branch-protection
+  branch_protection_rule:
+  pull_request:
+    branches: [ "main" ]
+
+# Declare default permissions as read only.
+permissions: read-all
+
+jobs:
+  analysis:
+    name: Scorecard analysis
+    runs-on: ubuntu-latest
+    permissions:
+      # Needed to upload the results to code-scanning dashboard.
+      security-events: write
+      # Needed to publish results and get a badge (see publish_results below).
+      id-token: write
+
+    steps:
+      - name: "Checkout code"
+        uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # v3.5.3
+        with:
+          persist-credentials: false
+
+      - name: "Run analysis"
+        uses: ossf/scorecard-action@e38b1902ae4f44df626f11ba0734b14fb91f8f86 # v2.1.2
+        with:
+          results_file: results.sarif
+          results_format: sarif
+          # (Optional) "write" PAT token. Uncomment the `repo_token` line below if:
+          # - you want to enable the Branch-Protection check on a *public* repository, or
+          # - you are installing Scorecard on a *private* repository
+          # To create the PAT, follow the steps in https://github.com/ossf/scorecard-action#authentication-with-pat.
+          # repo_token: ${{ secrets.SCORECARD_TOKEN }}
```

Comment:
I'm unclear here -- do we need the repo_token line or not? 

> - you want to enable the Branch-Protection check on a *public* repository, or

Reply:
I left the rep_token line because of this issue here: https://github.com/devfile/api/issues/1417

It is planned in the future to add the PAT so the scorecard correctly picks up on branch protection rules so I left that section commented for reference. If it's preferred or you don't want to add it for this repo I can remove it.

Reply:
That makes sense, it's just a slightly confusing doc -- to be clear we're currently _not_ enabling branch protection checks in the repo and so do not need the repo_token, correct?

I'm fine leaving it in there if the plan is to add it at some point in the future.

Reply:
Yes this PR is **not** going to have branch protection checks and the plan is to add all of those checks for Devfile repos in the future as part of the issue I linked

I can make the comment surrounding that token more clear?

Reply:
I don't think the comment is necessary, thanks for the clarification